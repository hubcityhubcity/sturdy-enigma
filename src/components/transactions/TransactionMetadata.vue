<template>
  <div class="tx__metadata">
    <p>
      Block
      <router-link
        v-if="checkFeatureAvailable()"
        :to="{ name: `block`, params: { height: transaction.height } }"
      >
        #{{ transaction.height | prettyInt }}</router-link
      >
      <span v-else>#{{ transaction.height | prettyInt }}</span>
      &nbsp;<i class="material-icons">access_time</i>&nbsp;{{ date }}
    </p>
    <p v-if="transaction.undelegationEndTime">
      Liquid date:
      {{ getUndelegationEndTime() }}
    </p>
    <p v-if="transaction.memo">
      <i class="material-icons">message</i> Memo: {{ transaction.memo }}
    </p>
    <p>
      Fee:
      <b>{{ transaction.fee.amount }}</b>
      <span> {{ transaction.fee.denom }}</span>
    </p>
    <p>
      Hash: <span class="hash">{{ transaction.hash }}</span>
    </p>
  </div>
</template>

<script>
import { mapGetters } from "vuex"
import gql from "graphql-tag"
import moment from "moment"
import { atoms, viewDenom } from "scripts/num.js"
import { prettyInt } from "scripts/num"

export default {
  name: `transaction-metadata`,
  filters: {
    atoms,
    viewDenom,
    prettyInt
  },
  props: {
    transaction: {
      type: Object,
      required: true
    }
  },
  data: () => ({
    network: {}
  }),
  computed: {
    ...mapGetters({ networkId: `network` }),
    date() {
      const momentTime = moment(this.transaction.timestamp)
      return momentTime.format(`HH:mm:ss`)
    }
  },
  methods: {
    getUndelegationEndTime() {
      return moment(new Date(this.transaction.undelegationEndTime))
    },
    checkFeatureAvailable() {
      const feature = `feature_explorer`
      return this.network[feature] === true
    }
  },
  apollo: {
    network: {
      query: gql`
        query NetworkActionModal($networkId: String!) {
          network(id: $networkId) {
            id
            feature_explorer
          }
        }
      `,
      variables() {
        /* istanbul ignore next */
        return {
          networkId: this.networkId
        }
      },
      update(data) {
        /* istanbul ignore next */
        return data.network
      }
    }
  }
}
</script>
<style scoped>
.material-icons {
  font-size: 1rem;
  vertical-align: middle;
  margin-bottom: 2px;
}

.hash {
  color: var(--dim);
  word-break: break-all;
}
</style>
