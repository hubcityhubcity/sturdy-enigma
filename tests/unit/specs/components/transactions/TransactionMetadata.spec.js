import { shallowMount } from "@vue/test-utils"
import TransactionMetadata from "src/components/transactions/TransactionMetadata"

describe(`TransactionMetadata`, () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(TransactionMetadata, {
      propsData: {
        timestamp: new Date("2019-07-31"),
        height: 1234567,
        fee: {
          amount: "0.001234",
          denom: "ATOM"
        }
      },
      stubs: [`router-link`]
    })
  })

  it(`renders correct fee and block transaction data`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
