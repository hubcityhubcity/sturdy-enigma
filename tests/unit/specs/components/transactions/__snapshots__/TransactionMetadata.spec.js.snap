// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionMetadata renders correct transaction metadata 1`] = `
<div
  class="tx__metadata"
>
  <p>
    
    Block
    
    <span>
      #1,234,567
    </span>
    
     
    <i
      class="material-icons"
    >
      access_time
    </i>
     00:00:00
  
  </p>
   
  <p>
    
    Liquid date:
    Sat Nov 16 2019 14:08:20 GMT+0000
  
  </p>
   
  <!---->
   
  <p>
    
    Fee:
    
    <b>
      0.001234
    </b>
     
    <span>
       ATOM
    </span>
  </p>
   
  <p>
    
    Hash: 
    <span
      class="hash"
    >
      ABCD1234
    </span>
  </p>
</div>
`;
